Introduce policy data wrapper class

jan-cerny opened the pull request:
#### Description:

Introduce a PolicyDataHandler class and move some items from the OSCAPScope class to this new class.

#### Rationale:
OSCAPScope is a class that handles GUI, but has too many other responsibilities, apart from GUI items there is many functional logic involved. The OSCAPScope often manipulates the PolicyData class, so it would be natural to move this code to the PolicyData class. However, this class is intended to work only as a data structure. We will not break this intention so we will create a new class PolicyDataHandler that manipulates the existing PolicyData class.

This separation also improves testability.

#### Review Hints:
all relates to GUI installation
./create_update_image.sh install_addon_from_repo
./reanaconda.py updates ~/work/git/oscap-anaconda-addon/update.img 
try to load content from a server, eg. provided by `python3 -m http.server`

jan-cerny committed:
Create a wrapper around the PolicyData class

The new class DataHandler will manipulate the PolicyData class.
It adds abstraction on the PolicyData and will allow to extract
modification of PolicyData out from the OSCAPSpoke.

jan-cerny committed:
Move methods from OSCAPSpoce to DataHandler

These 2 methods mainly modify the PolicyData, therefore the
DataHandler is a better place for them than the GUI.

jan-cerny committed:
Extract method set_url

Extract method set_url and move it to DataHandler because it mostly
manipulates with policy_data.

jan-cerny committed:
Rename DataHandler to PolicyDataHandler

jan-cerny committed:
Move content_defined method to PolicyDataHandler

This method is reading PolicyData, it isn't a property of the GUI.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/OpenSCAP/oscap-anaconda-addon/inspections/e400f52c-182f-4de9-bfa8-67e2e78cfb12) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/OpenSCAP/oscap-anaconda-addon/inspections/e400f52c-182f-4de9-bfa8-67e2e78cfb12) completed: **8 updated code elements**

marcusburghardt commented:
@matejak , could you help reviewing this PR, please? I believe you are more familiar with OAA. Thanks

matejak commented on org_fedora_oscap/gui/spokes/oscap.py:
```diff
@@ -402,7 +399,7 @@ def _fetch_data_and_initialize(self):
 
         thread_name = None
-        if self._policy_data.content_url and self._policy_data.content_type != "scap-security-guide":
+        if self.data_handler.needs_fetch_content():
```
I suggest to rename `needs_fetch_content` to `content_is_fetchable` or something like that, because the need arises from the fact that we don't have it, which may not be the case.

matejak commented:
This is great, it makes the code much more readable, and extracts reusable parts, so they can be reused, at least in theory.
Could you please check out whether this approach couldn't benefit functionality in https://github.com/OpenSCAP/oscap-anaconda-addon/blob/rhel9-branch/org_fedora_oscap/service/installation.py ? I think that the approach of fetching in https://github.com/OpenSCAP/oscap-anaconda-addon/blob/rhel9-branch/org_fedora_oscap/service/installation.py#L86 is wrong, and `needs_fetch_content` functionality could be useful here, because if we use system content, there is no need to even try to download anything. I think that now, some sort of fetching is attempted even in the case of the system content, and inevitable errors are just silently ignored.

jan-cerny committed:
Rename method

Rename needs_fetch_content to content_is_fetchable because the
need arises from the fact that we don't have it, which may not
be the case.

jan-cerny committed:
Improve condition for content fetching

In this situation, it doesn't make sense to attempt to fetch
content if the policy data says that the content isn't fetchable.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/OpenSCAP/oscap-anaconda-addon/inspections/92fe8918-2c3a-4086-9b5b-3f707943837e) was created.

jan-cerny commented:
I have renamed the aforementioned method and then I have changed the condition in PrepareValidContent.run().

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/OpenSCAP/oscap-anaconda-addon/inspections/92fe8918-2c3a-4086-9b5b-3f707943837e) completed: **9 updated code elements**

jan-cerny commented:
@matejak what else needs to be done?

matejak reviewed (approved):
Great, thanks!
